ledger/db/src/tx_out_store.rs

```diff
@@ -294,15 +294,11 @@ impl TxOutStore {
         let mut range_to_hash: HashMap<Range, [u8; 32]> = HashMap::default();
         for &(low, high) in ranges.iter().chain(other_ranges.iter()) {
             let range = Range::new(low, high)?;
-            let hash = if low >= num_tx_outs {
-                // Supply the nil hash if the range contains no data.
-                // Note: Nil hashes could probably be omitted as an optimization if validation
-                // knows that it must supply them for any range where `low >= num_tx_outs`.
-                *NIL_HASH
```

Comment:
Omitting this NIL_HASH when constructing a proof of membership is the main change.